bananas_api/new_upload/classifiers/heightmap.py

```diff
@@ -0,0 +1,76 @@
+from ...helpers.api_schema import Classification
+from ...helpers.enums import (
+    TerrainType,
+    Resolution,
+    Shape,
+)
+from ..readers.heightmap import Heightmap
+
+
+def classify_heightmap(heightmap: Heightmap) -> Classification:
+    classification = {}
+
+    surface = heightmap.size[0] * heightmap.size[1]
+    if surface < 512000:
+        classification["resolution"] = Resolution.STANDARD
+    else:
+        classification["resolution"] = Resolution.HIGH
+
+    aspect_ratio = heightmap.size[0] / heightmap.size[1]
+    if aspect_ratio < 1:
+        aspect_ratio = 1 / aspect_ratio
+
+    if aspect_ratio == 1:
+        classification["shape"] = Shape.SQUARE
+    elif aspect_ratio == 2:
+        classification["shape"] = Shape.RECTANGLE
+    else:
+        classification["shape"] = Shape.NARROW
+
+    # Change the histogram in small bins.
+    small_histogram = [0] * 16
+    for i, value in enumerate(heightmap.histogram):
+        # Skip sea-level.
+        if i == 0:
+            continue
+        small_histogram[i // 16] += value
+    # Normalize the histogram.
+    small_histogram = [value / (surface - heightmap.histogram[0]) for value in small_histogram]
+    classification["debug"] = ", ".join(["%.2f" % (value * 100) for value in small_histogram])
+
+    if any(value >= 0.9 for value in small_histogram):
+        classification["terrain-type"] = TerrainType.FLAT
+    elif len([value for value in small_histogram if value > 0.01]) >= 8:
```

Comment:
I prefer `sum(value > 0.01 for value in small_histogram) >= 8`. Easier to understand for me.